mergify_engine/web/__init__.py

```diff
@@ -358,6 +358,31 @@ async def queues(installation_id):
     return responses.JSONResponse(status_code=200, content=queues)
 
 
+@app.get(
+    "/queuesV2/{owner_id}",  # noqa: FS003
+    dependencies=[fastapi.Depends(auth.signature)],
+)
+async def queuesV2(owner_id):
+    queues = collections.defaultdict(dict)
+    async for queue in _AREDIS_CACHE.scan_iter(match=f"merge-queue~{owner_id}~*"):
```

Comment:
Isn't there a security hole there where you can request the queue for any repository if you know the owner id?

(this applies to the original method too, but it's harder since the installation id is not public)

Reply:
I don't think it's a big deal, since this endpoint is protected by `auth.signature`. This authentication method just trust that the dashboard has done the check first.

If one day we make this API public, we need to create a new authentication method that does the oauth flow on engine side that will validate the owner_id